bson/src/main/scala/DefaultBSONHandlers.scala

```diff
@@ -75,6 +78,23 @@ trait DefaultBSONHandlers {
     def write(date: Date) = BSONDateTime(date.getTime)
   }
 
+  implicit object BSONUUID extends BSONHandler[BSONBinary, UUID] {
```

Comment:
Rather as `BSONString`

Reply:
My goal was to be compatible with `UUID` from mongodb. If you open mongodb shell and do something like this:

```
> new UUID("00112233445566778899AABBCCDDEEFF")
```

you will notice that mongo will return:

```
BinData(3,"ABEiM0RVZneImaq7zN3u/w==")
```

That's why it's implemented this way.

Reply:
I doubt that's the expected behaviour in most of the case

Reply:
I think it is expected behavior for numerous reasons:
* Acording to [bson specification](http://bsonspec.org/spec.html) there is dedicated binary subtype for storing UUID so I think it is more idiomatic to store it this way
* UUID are very often used to search in database and mongodb indexes are optimized to handle certain array sizes (source: [Optimize Query Performance](https://docs.mongodb.com/v3.4/tutorial/optimize-query-performance-with-indexes-and-projections/). I haven't made any tests but I believe that it should be faster than text search.
* Obviously it uses less storage this way.
* Other people are doing this for example guys from PyMongo please read [this stackoverflow question](https://stackoverflow.com/questions/12030709/why-does-pymongo-encode-uuid-uuid1-as-a-bsonbinary)

Reply:
I'm not convinced it should be the default behaviour. You can find many project storing UUID as string, as in many case the binary optimization doesn't offer much improvement.

Reply:
From the people I could have asked about, most are storing it as BSON string, so for now I tend to have that string codec as default, and the binary one available but not the implicit scope by default.